snowpack/src/config.ts

```diff
@@ -887,20 +887,24 @@ export function validatePluginLoadResult(
   }
 }
 
-export function createConfiguration(
-  config: SnowpackUserConfig = {},
-): [ValidatorResult['errors'], undefined] | [null, SnowpackConfig] {
+class ConfigValidationError extends Error {
+  constructor(errors: Error[]) {
+    super(`Configuration Error:\n${errors.map((err) => `  - ${err.toString()}`).join('\n')}`);
```

Comment:
nit:
```suggestion
    super(`Configuration Error:\n${errors.map((err) => `  - ${err.toString()}`).join(os.EOL)}`);
```
https://nodejs.org/api/os.html#os_os_eol

Reply:
huh, we do `\n` in a lot of places... sounds like we should do a larger pass of this in a future PR?